### PR TITLE
[5486] Report socket connection issues on the UI via snacks

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5486

### CODE REVIEW

File: ui/app/src/state/epics/env.ts [hunk 1]
```diff
@@ -30,5 +40,25 @@ export default [
...
       )
+    ),
+  (action$, state$, { getIntl }) =>
+    action$.pipe(
+      ofType(setSiteSocketStatus.type),
+      filter(({ payload }) => !payload.connected),
+      tap(() => {
+        const hostToHost$ = getHostToHostBus();
+        hostToHost$.next(
+          showSystemNotification({
```
Comment (rart) [inline ui/app/src/state/epics/env.ts]: This needs to be persistent until the socket gets back online or otherwise dismissed so this doesn't work

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/state/epics/env.ts [hunk 2]
```diff
@@ -15,10 +15,20 @@
...
 import { catchAjaxError } from '../../utils/ajax';
 import { fetchSystemVersion, fetchSystemVersionComplete, fetchSystemVersionFailed } from '../actions/env';
+import { setSiteSocketStatus, showSystemNotification } from '../actions/system';
+import { getHostToHostBus } from '../../utils/subjects';
+import { defineMessages } from 'react-intl';
+
+const envMessages = defineMessages({
+  socketConnectionIssue: {
+    id: 'envMessages.socketConnectionIssue',
+    defaultMessage: 'Socket connection was interrupted. Studio will continue to retry the connection.'
```
Comment (rart) [inline ui/app/src/state/epics/env.ts]:
```suggestion
    defaultMessage: 'Connection with the server was interrupted. Studio will continue to retry the connection.'
```

**Review by rart — CHANGES_REQUESTED** [review]
We should:
- Disconnect the socket when session expires
- Reconnect when session resumes

File: ui/app/src/state/epics/env.ts (hunk 2 above)
Comment (jvega190) [reply ui/app/src/state/epics/env.ts]: When the session expires it does disconnect (and shows the snack).

File: ui/app/src/state/epics/auth.ts [hunk 3]
```diff
@@ -44,10 +44,19 @@ const epics: CrafterCMSEpic[] = [
...
     action$.pipe(
       ofType(loginComplete.type),
-      map(() => messageSharedWorker(refreshAuthToken()))
+      withLatestFrom(state$),
+      switchMap(([, state]) => {
+        const hasActiveSite = Boolean(state.sites.active);
+        return [
+          messageSharedWorker(refreshAuthToken()),
+          ...(hasActiveSite
+            ? [messageSharedWorker(openSiteSocket({ site: state.sites.active, xsrfToken: getXSRFToken() }))]
```
Comment (jvega190) [inline ui/app/src/state/epics/auth.ts]: Right now it opening the socket connection on login complete (not only re-login). Should it be only when logging in after a session expiration?

File: ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx [hunk 4]
```diff
@@ -1346,6 +1359,25 @@ export function PreviewConcierge(props: PropsWithChildren<{}>) {
           </IconButton>
         }
       />
+      <Snackbar
```
Comment (rart) [inline ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx:1364]: Make sure guest & socket snacks don't overlap

File: ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx [hunk 5]
```diff
@@ -365,6 +369,15 @@ export function PreviewConcierge(props: PropsWithChildren<{}>) {
     });
   };
 
+  useEffect(() => {
```
Comment (rart) [inline ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx:374]: Consider session status (should not see socket snack and re-login dialog)

File: ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx [hunk 6]
```diff
@@ -365,6 +369,15 @@ export function PreviewConcierge(props: PropsWithChildren<{}>) {
...
+    if (socketConnected) {
+      clearTimeout(socketConnectionTimeoutRef.current);
+      setSocketConnectionSnackbarOpen(false);
+    } else {
+      startGuestDetectionTimeout(socketConnectionTimeoutRef, setSocketConnectionSnackbarOpen);
```
Comment (rart) [inline ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx]: Rename to non-guest specific name

File: ui/app/src/state/epics/auth.ts [hunk 7]
```diff
@@ -44,10 +44,19 @@ const epics: CrafterCMSEpic[] = [
...
+            : [])
+        ];
```
Comment (rart) [inline ui/app/src/state/epics/auth.ts]:
```suggestion
        return [
          messageSharedWorker(refreshAuthToken()),
          // This action here assumes this epic is ran only when login from session timeout, not on initial login too
          hasActiveSite &&
            messageSharedWorker(openSiteSocket({ site: state.sites.active, xsrfToken: getXSRFToken() }))
        ].filter(Boolean);
```

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx (hunk 5 above)
Comment (rart) [reply ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx:374]: Keep the Snackbar state tied to a single prop: check authActive here.

File: ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx [hunk 8]
```diff
@@ -1346,6 +1361,26 @@ export function PreviewConcierge(props: PropsWithChildren<{}>) {
           </IconButton>
         }
       />
+      <Snackbar
+        open={socketConnectionSnackbarOpen && authActive}
```
Comment (rart) [inline ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx]: See above.

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**